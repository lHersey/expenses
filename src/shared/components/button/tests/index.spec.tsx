import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import Providers from 'shared/providers/test-providers';

import Button from '../index';

describe('<Button />', () => {
  describe('snapshots', () => {
    it('should match snapshot on base state', () => {
      expect.hasAssertions();
      const rendered = render(
        <Providers>
          <Button text="Hello Button" />
        </Providers>,
      ).toJSON();

      expect(rendered).toMatchSnapshot();
    });

    it('should match snapshot dense modifier', () => {
      expect.hasAssertions();
      const rendered = render(
        <Providers>
          <Button text="Hello Button" dense />
        </Providers>,
      ).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('unit', () => {
    it('should render the text', () => {
      expect.hasAssertions();
      const { getByText } = render(
        <Providers>
          <Button text="Hello Button" />
        </Providers>,
      );

      expect(getByText('Hello Button')).toBeTruthy();
    });

    it('should handle onPress event', () => {
      expect.hasAssertions();

      const handleOnPressMock = jest.fn();

      const { getByText } = render(
        <Providers>
          <Button text="Hello Button" onPress={handleOnPressMock} />
        </Providers>,
      );

      fireEvent.press(getByText('Hello Button'));

      expect(handleOnPressMock).toHaveBeenCalledWith();
    });
  });
});
