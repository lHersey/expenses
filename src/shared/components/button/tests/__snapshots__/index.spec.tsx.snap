// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshots should match snapshot dense modifier 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    dense={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 15,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minWidth": 100,
          "paddingBottom": 10,
          "paddingLeft": 7.5,
          "paddingRight": 7.5,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
        ]
      }
    >
      <Text
        dense={true}
        style={
          Array [
            Object {
              "color": "#5e5e5e",
              "fontFamily": "Inter_500",
              "fontSize": 16,
            },
            Object {
              "color": "#FFF",
            },
          ]
        }
      >
        Hello Button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`<Button /> snapshots should match snapshot on base state 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 15,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minWidth": 100,
          "paddingBottom": 20,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#5e5e5e",
              "fontFamily": "Inter_500",
              "fontSize": 16,
            },
            Object {
              "color": "#FFF",
            },
          ]
        }
      >
        Hello Button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
